### PR TITLE
Adds moffins, a whimsical breakfast pastry

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request
Athath wanted to pr a cute idea of adding cloth to a muffin recipe so moths would like them more, but let that pr stale out so I took the idea and made it funnier.

Mobs that like eating cloth get a cute message about being excited, otherwise you get a message expressing doubt.
Adds cloth as a disgusted food for the other races, because it's funny and adds to the meme.

image

## Why It's Good For The Game
image
Chefs can make a joke of sprinkling cloth on their muffins as a treat for moths, which other crewmembers might not find so appealing... It picks one of the three icons at random, so there's some variance to the sprinkled "goods"!

## Changelog
:cl:
add: Chefs can now cook moffins! Yaaay moffins! Check the pastries subsection of the crafting menu to make some! Cutting up gauze and bedsheets can get you started!
/:cl:

### CODE REVIEW

File: code/game/objects/items/food/pastries.dm [hunk 1]
```diff
@@ -357,7 +357,26 @@
...
+/obj/item/food/muffin/moffin/Initialize(mapload)
+	. = ..()
+	icon_state = "[icon_state]_[rand(1,3)]"
 
+/obj/item/food/muffin/moffin/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/moffin_observer = user
+	if(foodtypes & moffin_observer.dna.species.liked_food)
```
Comment (Mothblocks) [inline code/game/objects/items/food/pastries.dm:376]: I don't think this is correct--if someone who likes grain, sugar, or breakfast looks at it they'll still get this text. I think you just want to check `liked_food & CLOTH`.

File: code/game/objects/items/food/pastries.dm [hunk 2]
```diff
@@ -357,7 +357,26 @@
...
+/obj/item/food/muffin/moffin/Initialize(mapload)
+	. = ..()
+	icon_state = "[icon_state]_[rand(1,3)]"
 
+/obj/item/food/muffin/moffin/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/moffin_observer = user
+	if(CLOTH & moffin_observer.dna.species.liked_food)
```
Comment (Mothblocks) [inline code/game/objects/items/food/pastries.dm:376]:
```suggestion
	if(moffin_observer.dna.species.liked_food & CLOTH)
```

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use

Comment (ArcaneDefence) [reply code/game/objects/items/food/pastries.dm:376]: image

**Review by Mothblocks — APPROVED** [review]
Code ✔

File: code/game/objects/items/food/pastries.dm [hunk 3]
```diff
@@ -357,7 +357,26 @@
...
 
+/obj/item/food/muffin/moffin/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/moffin_observer = user
+	if(moffin_observer.dna.species.liked_food & CLOTH)
+		. += "<span class='nicegreen'>Ooh! It's even got bits of clothes on it! Yummy!</span>"
+	else
+		. += "<span class='warning'>I'm not too sure what's on top though...</span>"
```
Comment (ATH1909) [inline code/game/objects/items/food/pastries.dm:379]:
```suggestion
		. += "<span class='warning'>You're not too sure what's on top though...</span>"
```

**Comment by ATH1909** [comment]
❤️ ❤️ ❤️

**Pull request merged**